simulation/halsim_lowfi/src/main/native/cpp/NTProvider_Relay.cpp

```diff
@@ -0,0 +1,27 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2017 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "NTProvider_Relay.h"
+
+#include <HAL/Ports.h>
+#include <MockData/RelayData.h>
+
+void HALSimNTProviderRelay::Initialize() {
+  InitializeDefault(HAL_GetNumRelayHeaders(), HALSIM_RegisterRelayAllCallbacks);
+}
+
+void HALSimNTProviderRelay::OnCallback(
+    uint32_t chan, std::shared_ptr<nt::NetworkTable> table) {
+  table->GetEntry("init_fwd?")
+      .SetBoolean(HALSIM_GetRelayInitializedForward(chan));
+  table->GetEntry("init_rvs?")
+      .SetBoolean(HALSIM_GetRelayInitializedReverse(chan));
+  table->GetEntry("fwd?").SetBoolean(HALSIM_GetRelayForward(chan));
+  table->GetEntry("rvs?").SetBoolean(HALSIM_GetRelayReverse(chan));
+
+  table->GetInstance().Flush();
```

Comment:
Not sure I'm comfortable with all of these flushes. It's a good way to get some weird synchrony during the sends. I almost would recommend for this just setting update rate to 10ms and not doing the flushes.

Reply:
With some testing I've noticed that without the flushes NT drags behind substantially, sometimes not updating for a few seconds. The idea here is that the changeable update rate should mitigate the use of more flushes.

Reply:
As an aside, the flush rate depends highly on other extensions that call the main driver station notify function (in this case, this is a different PR, #829 ). Do you think it's worth creating a background thread in this extension to flush every 'x' hz?

Reply:
That's weird that nt falls behind. Manually having a thread to call flush every 20ms at the low level is exactly what setting an update rate of 20ms in nt does. They're basically the same code. So I'd rather figure out why that is lagging rather then a new thread

Reply:
I'll look more into it, it may have easily been an issue with Shuffleboard ghosting

Reply:
https://github.com/wpilibsuite/shuffleboard/issues/316